echopype/clean/utils.py

```diff
@@ -83,6 +84,95 @@ def calc_transient_noise_pooled_Sv(
     return pooled_Sv
 
 
+def index_binning_pool_Sv(
+    ds_Sv: xr.Dataset,
+    func: str,
+    depth_bin: int,
+    num_side_pings: int,
+    exclude_above: float,
+    chunk_dict: dict,
+) -> xr.DataArray:
+    """
+    Compute pooled Sv array for transient noise masking using index binning.
+
+    This function makes the assumption that within each channel, the difference
+    between depth values is uniform across all pings. Thus, computing the number of
+    range sample indices needed to cover the depth bin is a channel-specific task.
+    """
+    # Create `ds_Sv` copy
+    ds_Sv_copy = (
+        ds_Sv.copy()
+        # Drop `filenames` dimension if exists and transpose Dataset
+        .drop_dims("filenames", errors="ignore").transpose("channel", "ping_time", "range_sample")
+    )
+
+    # Compute number of range sample indices that are needed to encapsulate the `depth_bin`
+    # value per channel.
+    all_chan_num_range_sample_indices = np.ceil(
+        depth_bin / np.nanmean(np.diff(ds_Sv_copy["depth"], axis=2), axis=(1, 2))
+    ).astype(int)
+
+    # Create list for pooled Sv DataArrays
+    pooled_Sv_list = []
+
+    # Iterate through channels
+    for channel_index in range(len(ds_Sv_copy["channel"])):
+        # Create calibrated Sv DataArray copies and remove values too close to the surface
+        min_range_sample = (ds_Sv["depth"] <= exclude_above).argmin().values
+        chan_Sv = (
+            ds_Sv_copy["Sv"]
+            .copy()
+            .isel(
+                channel=channel_index,
+                range_sample=slice(min_range_sample, len(ds_Sv["range_sample"])),
+            )
+        )
+        chan_pooled_Sv = (
+            ds_Sv_copy["Sv"]
+            .copy()
+            .isel(
+                channel=channel_index,
+                range_sample=slice(min_range_sample, len(ds_Sv["range_sample"])),
+            )
+        )
```

Comment:
See my comment above. Also for the beginning or ending parts you can just use `None` in `slice`.

Reply:
Added this suggestion